app/controlplane/internal/service/attestation.go

```diff
@@ -186,34 +189,75 @@ func (s *AttestationService) Init(ctx context.Context, req *cpAPI.AttestationSer
 }
 
 func (s *AttestationService) Store(ctx context.Context, req *cpAPI.AttestationServiceStoreRequest) (*cpAPI.AttestationServiceStoreResponse, error) {
+	var envelope dsse.Envelope
 	robotAccount := usercontext.CurrentRobotAccount(ctx)
 	if robotAccount == nil {
 		return nil, errors.NotFound("not found", "robot account not found")
 	}
 
-	// This will make sure the provided workflowRunID belongs to the org encoded in the robot account
-	wf, err := s.findWorkflowFromTokenOrNameOrRunID(ctx, robotAccount.OrgID, "", "", req.WorkflowRunId)
+	// Try unmarshaling a bundle first, falling back to plain dsse envelopes
+	var bundle protobundle.Bundle
+	if req.GetBundle() != nil {
+		if err := protojson.Unmarshal(req.GetBundle(), &bundle); err != nil {
+			return nil, handleUseCaseErr(err, s.log)
+		}
+		envelope = *envelopeFromBundle(&bundle)
+	} else {
+		// trying with envelope instead
+		if err := json.Unmarshal(req.GetAttestation(), &envelope); err != nil {
+			return nil, handleUseCaseErr(err, s.log)
+		}
+	}
+
+	digest, err := s.storeAttestation(ctx, &envelope, robotAccount, req.WorkflowRunId, req.MarkVersionAsReleased)
 	if err != nil {
 		return nil, handleUseCaseErr(err, s.log)
 	}
 
-	// Decode the envelope through json encoding but
-	// TODO: Verify the envelope signature before storing it
-	// see sigstore's dsse signer/verifier helpers
-	envelope := &dsse.Envelope{}
-	if err := json.Unmarshal(req.Attestation, envelope); err != nil {
-		return nil, handleUseCaseErr(err, s.log)
+	// save bundle after saving attestation
+	if bundle.GetDsseEnvelope() != nil {
+		_, err = s.wrUseCase.SaveBundle(ctx, req.WorkflowRunId, &bundle)
```

Comment:
I have the feeling we shoudl have used the same storeAttestation use-case but it can receive either a bundle or an attestation. 
note that you might be storing the bundle before we store the older attestation for example. Also, are all the authorization checks done in this case too? i.e can't save a bundle for a workflow you don't own?

Reply:
Implemented in the same use case.